From 9a0039e46d546ea6f92c644f0dc2c7a394a9a575 Mon Sep 17 00:00:00 2001
From: Andreas Mohr <and@gmx.li>
Date: Tue, 8 Jan 2019 08:06:54 +0000
Subject: [PATCH] autoconf: Cleanup AC_TRY_LINK obsolete warning

configure.ac:149: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:262: warning: The macro `AC_TRY_LINK' is obsolete.

Signed-off-by: Andreas Mohr <and@gmx.li>
---
 m4.include/mc-get-fs-info.m4         | 6 +++---
 m4.include/mc-use-termcap.m4         | 6 +++---
 m4.include/mc-with-screen-ncurses.m4 | 8 ++++----
 3 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/m4.include/mc-get-fs-info.m4 b/m4.include/mc-get-fs-info.m4
index 8b61ef296..bc96e99ee 100644
--- a/m4.include/mc-get-fs-info.m4
+++ b/m4.include/mc-get-fs-info.m4
@@ -5,14 +5,14 @@ dnl
 
 AC_DEFUN([gl_POSIX_FALLOCATE], [
     dnl * Old glibcs have broken posix_fallocate(). Make sure not to use it.
-    AC_TRY_LINK([
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
         #define _XOPEN_SOURCE 600
         #include <stdlib.h>
         #if defined(__GLIBC__) && (__GLIBC__ < 2 || __GLIBC_MINOR__ < 7)
             possibly broken posix_fallocate
         #endif
-    ],
-    [posix_fallocate(0, 0, 0);],
+    ]],
+    [[posix_fallocate(0, 0, 0);]])],
     [AC_DEFINE(
         [HAVE_POSIX_FALLOCATE],
         [1],
diff --git a/m4.include/mc-use-termcap.m4 b/m4.include/mc-use-termcap.m4
index b37f7496b..11f4a94a8 100644
--- a/m4.include/mc-use-termcap.m4
+++ b/m4.include/mc-use-termcap.m4
@@ -17,14 +17,14 @@ AC_DEFUN([mc_SLANG_TERMCAP], [
     AC_CACHE_CHECK([if S-Lang uses termcap], [mc_cv_slang_termcap], [
 	ac_save_LIBS="$LIBS"
 	LIBS="$LIBS -lslang"
-	AC_TRY_LINK([
+	AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #ifdef HAVE_SLANG_SLANG_H
 #include <slang/slang.h>
 #else
 #include <slang.h>
 #endif
-		    ],
-		    [SLtt_get_terminfo(); SLtt_tgetflag((char*)"");],
+		    ]],
+		    [[SLtt_get_terminfo(); SLtt_tgetflag((char*)"");]])],
 		    [mc_cv_slang_termcap=no],
 		    [mc_cv_slang_termcap=yes])
 	LIBS="$ac_save_LIBS"
diff --git a/m4.include/mc-with-screen-ncurses.m4 b/m4.include/mc-with-screen-ncurses.m4
index 096192fac..80d8478c9 100644
--- a/m4.include/mc-with-screen-ncurses.m4
+++ b/m4.include/mc-with-screen-ncurses.m4
@@ -143,10 +143,10 @@ AC_DEFUN([mc_WITH_NCURSES], [
     dnl check for ESCDELAY
     AC_CACHE_CHECK([for ESCDELAY variable],
                    [mc_cv_ncurses_escdelay],
-                   [AC_TRY_LINK([], [
+                   [AC_LINK_IFELSE([AC_LANG_PROGRAM([], [[
                         extern int ESCDELAY;
                         ESCDELAY = 0;
-                        ],
+                        ]])],
                         [mc_cv_ncurses_escdelay=yes],
                         [mc_cv_ncurses_escdelay=no])
     ])
@@ -194,10 +194,10 @@ AC_DEFUN([mc_WITH_NCURSESW], [
 
     AC_CACHE_CHECK([for ESCDELAY variable],
 		   [mc_cv_ncursesw_escdelay],
-		   [AC_TRY_LINK([], [
+		   [AC_LINK_IFELSE([AC_LANG_PROGRAM([], [[
 			extern int ESCDELAY;
 			ESCDELAY = 0;
-			],
+			]])],
 			[mc_cv_ncursesw_escdelay=yes],
 			[mc_cv_ncursesw_escdelay=no])
     ])
